hammerblade/torch/kernel/bsg_tensor.hpp

```diff
@@ -13,4 +31,100 @@ typedef struct {
 #endif
 } bsg_tensor_t;
```

Comment:
Actually, I think I will add a new data structure for offloading a TensorIterator.